modules/distribution/resources/migration-1.8.0_to_1.9.0/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/internal/APIMMigrationServiceComponent.java

```diff
@@ -0,0 +1,249 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.wso2.carbon.apimgt.migration.client.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.carbon.apimgt.impl.APIManagerConfigurationService;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+import org.wso2.carbon.apimgt.migration.APIMigrationException;
+import org.wso2.carbon.apimgt.migration.client.MigrateFrom18to19;
+import org.wso2.carbon.apimgt.migration.client.MigrationClient;
+import org.wso2.carbon.apimgt.migration.util.Constants;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.registry.core.service.TenantRegistryLoader;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * @scr.component name="org.wso2.carbon.apimgt.migration.client" immediate="true"
+ * @scr.reference name="realm.service"
+ * interface="org.wso2.carbon.user.core.service.RealmService" cardinality="1..1"
+ * policy="dynamic" bind="setRealmService" unbind="unsetRealmService"
+ * @scr.reference name="registry.service"
+ * interface="org.wso2.carbon.registry.core.service.RegistryService" cardinality="1..1"
+ * policy="dynamic" bind="setRegistryService" unbind="unsetRegistryService"
+ * @scr.reference name="registry.core.dscomponent"
+ * interface="org.wso2.carbon.registry.core.service.RegistryService" cardinality="1..1"
+ * policy="dynamic" bind="setRegistryService" unbind="unsetRegistryService"
+ * @scr.reference name="tenant.registryloader" interface="org.wso2.carbon.registry.core.service.TenantRegistryLoader" cardinality="1..1"
+ * policy="dynamic" bind="setTenantRegistryLoader" unbind="unsetTenantRegistryLoader"
+ * @scr.reference name="apim.configuration" interface="org.wso2.carbon.apimgt.impl.APIManagerConfigurationService" cardinality="1..1"
+ * policy="dynamic" bind="setApiManagerConfig" unbind="unsetApiManagerConfig"
+ */
+
+@SuppressWarnings("unused")
+public class APIMMigrationServiceComponent {
+
+    private static final Log log = LogFactory.getLog(APIMMigrationServiceComponent.class);
+
+    /**
+     * Method to activate bundle.
+     *
+     * @param context OSGi component context.
+     */
+    protected void activate(ComponentContext context) {
+        String migrateVersion = null;
+        boolean cleanupNeeded = false;
+        boolean isDBMigrationNeeded = false;
+        boolean isRegistryMigrationNeeded = false;
+        boolean isFileSystemMigrationNeeded = false;
+
+        try {
+            APIMgtDBUtil.initialize();
+        } catch (Exception e) {
```

Comment:
Use proper multi-catch statements instead of catching generic exceptions. Also, log the exception trace.